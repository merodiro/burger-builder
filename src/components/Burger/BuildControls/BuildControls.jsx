import React from 'react'
import PropTypes from 'prop-types'

import classes from './BuildControls.module.css'
import BuildControl from './BuildControl/BuildControl'

const BuildControls = props => {
  const controls = [
    { label: 'Salad', type: 'salad' },
    { label: 'Bacon', type: 'bacon' },
    { label: 'Cheese', type: 'cheese' },
    { label: 'Meat', type: 'meat' },
  ]
  return (
    <div className={classes.BuildControls}>
      <p>
        Current Price: <strong>{props.price.toFixed(2)}</strong>
      </p>
      {controls.map(ctrl => (
        <BuildControl
          key={ctrl.label}
          label={ctrl.label}
          added={() => props.ingredientAdded(ctrl.type)}
          removed={() => props.ingredientRemoved(ctrl.type)}
          disabled={props.disabled[ctrl.type]}
        />
      ))}
      <button
        className={classes.OrderButton}
        disabled={!props.purchasable}
        onClick={props.ordered}
      >
        Order Now
      </button>
    </div>
  )
}

BuildControls.propTypes = {
  price: PropTypes.number.isRequired,
  purchasable: PropTypes.bool.isRequired,
  ingredientAdded: PropTypes.func,
  ingredientRemoved: PropTypes.func,
  ordered: PropTypes.func,
  disabled: PropTypes.object.isRequired,
}

export default BuildControls
