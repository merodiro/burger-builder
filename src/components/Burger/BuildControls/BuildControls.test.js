import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import BuildControls from './BuildControls'

describe('<BuildControls>', () => {
  it('renders the same output', () => {
    let disabled = {
      bacon: true,
      salad: false,
      meat: true,
      cheese: true
    }
    const component = shallow(
      <BuildControls price={4} disabled={{ bacon: true }} />
    )
    expect(component).toMatchSnapshot()
  })
})
