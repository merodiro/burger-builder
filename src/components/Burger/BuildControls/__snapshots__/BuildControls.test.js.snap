// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildControls> renders the same output 1`] = `
<div
  className="BuildControls"
>
  <p>
    Current Price: 
    <strong>
      4.00
    </strong>
  </p>
  <BuildControl
    added={[Function]}
    key="Salad"
    label="Salad"
    removed={[Function]}
  />
  <BuildControl
    added={[Function]}
    disabled={true}
    key="Bacon"
    label="Bacon"
    removed={[Function]}
  />
  <BuildControl
    added={[Function]}
    key="Cheese"
    label="Cheese"
    removed={[Function]}
  />
  <BuildControl
    added={[Function]}
    key="Meat"
    label="Meat"
    removed={[Function]}
  />
</div>
`;
